cypress/componentTheme/themes.test.js

```diff
@@ -0,0 +1,322 @@
+import React from "react";
+import Button from "../../src/components/button/button.component";
+import Link from "../../src/components/link/link.component";
+import Loader from "../../src/components/loader/loader.component";
+import LoaderBar from "../../src/components/loader-bar/loader-bar.component";
+import MultiActionButton from "../../src/components/multi-action-button/multi-action-button.component";
+import Pill from "../../src/components/pill/pill.component";
+import SplitButton from "../../src/components/split-button/split-button.component";
+import StepSequenceItem from "../../src/components/step-sequence/step-sequence-item/step-sequence-item.component";
+import { Tabs, Tab } from "../../src/components/tabs";
+
+import {
+  sageTheme,
+  mintTheme,
+  noTheme,
+  aegeanTheme,
+} from "../../src/style/themes/index";
+import { buttonDataComponent } from "../locators/button";
+import CypressMountWithProviders from "../support/component-helper/cypress-mount";
+import {
+  linkComponent,
+  loaderBarComponent,
+  loaderComponent,
+} from "../locators/themes";
+import { pillPreview } from "../locators/pill";
+import { stepSequenceDataComponentItem } from "../locators/step-sequence";
+import { getDataElementByValue } from "../locators";
+import { multiActionButtonComponent } from "../locators/multi-action-button";
+import { splitMainButton } from "../locators/split-button";
+
+const commonColors = [
+  ["mint", "rgb(0, 125, 90)"],
+  ["aegean", "rgb(0, 115, 194)"],
+  ["none", "rgb(0, 130, 0)"],
+  ["sage", "rgb(0, 126, 69)"],
+];
+
+const commonColorsOnHover = [
+  ["mint", "rgb(0, 96, 70)"],
+  ["aegean", "rgb(0, 92, 154)"],
+  ["none", "rgb(0, 99, 0)"],
+  ["sage", "rgb(0, 103, 56)"],
+];
+
+const multiButtonOnHover = [
+  ["mint", "rgb(0, 64, 46)"],
+  ["aegean", "rgb(0, 68, 114)"],
+  ["none", "rgb(0, 77, 42)"],
+  ["sage", "rgb(0, 77, 42)"],
+];
+
+const stepSequnceColors = [
+  ["mint", "rgb(0, 125, 90)"],
+  ["aegean", "rgb(0, 115, 194)"],
+  ["none", "rgb(0, 138, 33)"],
+  ["sage", "rgb(0, 138, 33)"],
+];
+
+const loaderBarColors = [
+  ["mint", "rgb(179, 227, 214)"],
+  ["aegean", "rgb(179, 214, 239)"],
+  ["none", "rgb(179, 224, 179)"],
+  ["sage", "rgb(179, 217, 200)"],
+];
+
+context("Testing Themes", () => {
+  describe.each([
```

Comment:
As each theme already stores the `name` property I don't think we should store them in the arrays as well, you could write a function that builds the test arrays:

```
const commonColors = [
  "rgb(0, 125, 90)",
  "rgb(0, 115, 194)",
  "rgb(0, 130, 0)",
  "rgb(0, 126, 69)",
];

const themes = [mintTheme, aegeanTheme, noTheme, sageTheme];

const buildTestArray = (array) => themes.map((theme, i) => ([theme.name, theme, array[i]]));

describe.each(buildTestArray(commonColors))("should render components with %s theme", (themeName, theme, color) => {
   CypressMountWithProviders(<Button>Small</Button>, theme);
   buttonDataComponent().should("have.css", "color", color);
```

Reply:
@edleeks87 thanks for this comment, applied as a valuable one :)